src/components/Form/story.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import Form from '.';
+import Readme from './README.md';
+
+storiesOf('Uncategorized/Form', module)
+  .addParameters({
+    readme: {
+      sidebar: Readme,
+    },
+  })
+  .add(
+    'Overview',
+    () => (
+      <Form
+        initialValues={{ nameInput: 'james r. cat' }}
+        // eslint-disable-next-line no-alert
+        onSubmit={(formValues) => alert(JSON.stringify(formValues))}
+        validationSchema={{
+          nameInput: {
+            message: 'No empty strings',
+            validate: (value) => value !== '',
+          },
+        }}
+      >
+        {({
+          formState,
```

Comment:
If you combined some parts of `formState` and `formValues`, could you do something like:
```
<input
  value={form.email.value}
  onChange={form.email.onChange}
  data-touched={form.email.touched}  
  data-error-messages={form.email.errorMessage}  
/>
```

Or, if you wanted to extract some properties in the function as child argument, you could get as simple as:
```
<Form ... />
  {
    ({ form: { touched, errorMessage, ...email } }) => {
      <input {...email}  />
    }
  }
</Form>
```